### PR TITLE
Enable single cell plot with two single cell gene expression terms; other improvements

### PR DESCRIPTION
**Pull request by creilly8**
# Description
This pull request enhances the handling and grouping of subplots in single cell plot visualizations, improves test coverage for subplot naming (especially for image plots), and ensures more robust handling of sample and plot metadata across the UI. Key changes include centralizing plot name logic, updating tests, and improving metadata extraction and configuration handling. 

Major changes:

1. Added support for passing coordinate term wrappers (`coordTWs`) in the single cell scatter model to support advanced plotting scenarios. This specifically focuses on scge terms. Note: This logic will need to be enhanced once numeric scct-like terms are added in memory.  

Minor changes:

1. Centralized plot name resolution by moving `getPlotName` logic from `SectionRenderer` to `SubplotManager`, and updated all usages and tests to call through the manager. This ensures consistent naming, including for special cases like `imagePlot`. Updated `SectionRenderer`'s `getKey` method to use the centralized `getPlotName` and accept the subplot manager as a parameter, ensuring correct grouping by plot name. Added and updated unit tests in `SectionRenderer.unit.spec.ts` and `SubplotManager.unit.spec.ts` to cover new plot naming logic, especially for `imagePlot` types and edge cases.
2. Refactored test helpers to ensure subplot manager is always initialized in tests that require grouping or naming logic.
3.  Improved sample and plot metadata extraction, including handling cases where the first column label in the sample table should be blank for meta results
4. Updated `getSingleCellSpecialCase` utility to accept a `noCohortTerms` flag and clarified documentation and return types for better downstream handling
5. Fixed issue with scge scatter plots launched from the summary not using the single cell scatter model. 

These changes improve consistency, maintainability, and test coverage for subplot naming and grouping logic in the single cell plotting codebase. This PR supports the ongoing efforts detailed in the [sc roadmap](https://github.com/stjude/sjpp/issues). 

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.

- [x] Tests: Added and/or passed unit and integration tests, or N/A
- [x] Todos: Commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A
- [x] Rust: Checked to see whether Rust needs to be re-compiled because of this PR, or N/A

### CODE REVIEW

File: server/src/singleCell/plotsRoute.ts [hunk 1]
```diff
@@ -51,21 +52,51 @@ function validTermdbSingleCellPlotsRequest(input): TermdbSingleCellPlotsRequest
...
+					return input.colorTW as TermWrapper
+			  })()
+			: undefined,
+		coordTWs: input?.coordTWs?.length
+			? (() => {
+					if (!isSingleCellTerm(input.coordTWs[0].term))
+						throw new Error('coordTWs must be an array of single cell terms for single cell scatter plot')
+					return input.coordTWs as TermWrapper[]
+			  })()
+			: undefined
```
[the remark this answers is not in the document]
Comment (creilly8) [reply server/src/singleCell/plotsRoute.ts:99]: No. The second term can be a cohort level, numeric term, such as 'Age'. Only `config.term` needs to be validated as a single cell term.

File: server/src/singleCell/plotsRoute.ts [hunk 2]
```diff
@@ -92,84 +123,42 @@ export function init({ genomes }) {
 
 async function getSingleCellScatter(req, res, ds) {
 	const q = req.query as TermdbSingleCellPlotsRequest
-	const { name, sample } = q.singleCellPlot
+
+	if (q.coordTWs?.length && q.colorTW) {
+		throw new Error('Using coordTWs with colorTW is not implemented for single cell scatter plot')
+	}
```
[the remark this answers is not in the document]
Comment (creilly8) [reply server/src/singleCell/plotsRoute.ts:129]: No, this whole function is in a try catch block. It returns fine.

File: server/src/singleCell/plotsRoute.ts [hunk 3]
```diff
@@ -215,55 +207,125 @@ async function getSingleCellScatter(req, res, ds) {
...
+			const formattedCell = {
+				sampleId: cell.sample,
+				x: cell.x,
+				y: cell.y,
+				z: 0,
+				hidden: { category: false },
+				category,
+				shape: cell.shape || 'Ref',
+				geneExp: cell.geneExp
+			}
```
[the remark this answers is not in the document]
Comment (creilly8) [reply server/src/singleCell/plotsRoute.ts:289]: This shouldn't be necessary but committing to avoid reinvestigating over and over.

**Comment by xzhou82** [comment]
minor issue. at [sc umap](http://localhost:3000/?mass={%22dslabel%22:%22GDC%22,%22genome%22:%22hg38%22,%22nav%22:{%22header_mode%22:%22hidden%22},%22plots%22:[{%22chartType%22:%22sampleScatter%22,%22singleCellPlot%22:{%22name%22:%22UMAP%22,%22sample%22:{%22sID%22:%22C3N-03186-01%22,%22eID%22:%22af20778d-911c-45ee-a725-573be122d8b2%22}},%22colorTW%22:{%22term%22:{%22name%22:%22Cluster%22,%22plot%22:%22UMAP%22,%22type%22:%22singleCellCellType%22,%22sample%22:{%22sID%22:%22C3N-03186-01%22,%22eID%22:%22af20778d-911c-45ee-a725-573be122d8b2%22}}}}]}), on replacing colorTW the termtypesearch correctly shows sc terms; then delete colorTW and click tw prompt, termtypesearch now shows cohort terms

**Review by xzhou82 — APPROVED** [review]
